-------------------------------------------------------------------------------
Test set: org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator
-------------------------------------------------------------------------------
Tests run: 15, Failures: 3, Errors: 0, Skipped: 0, Time elapsed: 19.191 s <<< FAILURE! - in org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator
testAuthenticationHttpClientPostWithMultiAuthHandler(org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator)  Time elapsed: 1.681 s  <<< FAILURE!
java.lang.AssertionError: expected:<200> but was:<401>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.apache.hadoop.security.authentication.client.AuthenticatorTestCase.doHttpClientRequest(AuthenticatorTestCase.java:236)
	at org.apache.hadoop.security.authentication.client.AuthenticatorTestCase._testAuthenticationHttpClient(AuthenticatorTestCase.java:246)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator$6.call(TestKerberosAuthenticator.java:226)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator$6.call(TestKerberosAuthenticator.java:223)
	at org.apache.hadoop.security.authentication.KerberosTestUtils$1.run(KerberosTestUtils.java:120)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:533)
	at org.apache.hadoop.security.authentication.KerberosTestUtils.doAs(KerberosTestUtils.java:117)
	at org.apache.hadoop.security.authentication.KerberosTestUtils.doAsClient(KerberosTestUtils.java:133)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator.testAuthenticationHttpClientPostWithMultiAuthHandler(TestKerberosAuthenticator.java:223)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1589)

testAuthenticationHttpClient(org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator)  Time elapsed: 1.114 s  <<< FAILURE!
java.lang.AssertionError: expected:<200> but was:<401>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.apache.hadoop.security.authentication.client.AuthenticatorTestCase.doHttpClientRequest(AuthenticatorTestCase.java:236)
	at org.apache.hadoop.security.authentication.client.AuthenticatorTestCase._testAuthenticationHttpClient(AuthenticatorTestCase.java:246)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator$3.call(TestKerberosAuthenticator.java:164)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator$3.call(TestKerberosAuthenticator.java:161)
	at org.apache.hadoop.security.authentication.KerberosTestUtils$1.run(KerberosTestUtils.java:120)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:533)
	at org.apache.hadoop.security.authentication.KerberosTestUtils.doAs(KerberosTestUtils.java:117)
	at org.apache.hadoop.security.authentication.KerberosTestUtils.doAsClient(KerberosTestUtils.java:133)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator.testAuthenticationHttpClient(TestKerberosAuthenticator.java:161)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1589)

testAuthenticationHttpClientPost(org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator)  Time elapsed: 1.098 s  <<< FAILURE!
java.lang.AssertionError: expected:<200> but was:<401>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.apache.hadoop.security.authentication.client.AuthenticatorTestCase.doHttpClientRequest(AuthenticatorTestCase.java:236)
	at org.apache.hadoop.security.authentication.client.AuthenticatorTestCase._testAuthenticationHttpClient(AuthenticatorTestCase.java:246)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator$4.call(TestKerberosAuthenticator.java:178)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator$4.call(TestKerberosAuthenticator.java:175)
	at org.apache.hadoop.security.authentication.KerberosTestUtils$1.run(KerberosTestUtils.java:120)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:533)
	at org.apache.hadoop.security.authentication.KerberosTestUtils.doAs(KerberosTestUtils.java:117)
	at org.apache.hadoop.security.authentication.KerberosTestUtils.doAsClient(KerberosTestUtils.java:133)
	at org.apache.hadoop.security.authentication.client.TestKerberosAuthenticator.testAuthenticationHttpClientPost(TestKerberosAuthenticator.java:175)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1589)

